### PR TITLE
Fixes to get OCaml CI passing

### PR DESCRIPTION
**Pull request by punchagan**

### CODE REVIEW

**Comment by punchagan** [comment]
Looks like the request to approve adding this repo to the OCaml CI GitHub app wasn't yet accepted. But, the build results for this PR are availeble at https://ocaml.ci.dev/github/punchagan/opam-ci-check/commit/fec7863fb3103feced1d1106992a140b3e3a0ef6

File: test/scripts/setup_sources.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env sh
+
+# Generate a tarball from the given files and update the sha256 in the opam file
+
+set -eu
+
```
Comment (shonfeder) [inline test/scripts/setup_sources.sh:6]: I suggest also adding

```suggestion
set -eu
set -o pipefail
```

**Review by shonfeder — APPROVED** [review]
Nice solution!

LGTM

File: test/scripts/setup_sources.sh (hunk 1 above)
Comment (punchagan) [reply test/scripts/setup_sources.sh:6]: Dash doesn't seem to have `pipefail`, but `bash` does. I'm not sure if we should just switch to running the script with `bash`, given all the OSes and architectures we run the CI on. I'm leaning towards making this a follow-up, and merging this PR for now.

Comment (shonfeder) [reply test/scripts/setup_sources.sh:6]: Oh I misssed this was dash! I'm not familiar with it and maybe it doesn't need pipefail. No worries here :) review oversight in my part.

**Pull request merged**